src/cargo/ops/cargo_new.rs

```diff
@@ -866,15 +867,115 @@ mod tests {
             }
         }
     }
-
-    if let Err(e) = Workspace::new(&path.join("Cargo.toml"), config) {
-        crate::display_warning_with_error(
-            "compiling this new package may not work due to invalid \
+    match Workspace::new(&path.join("Cargo.toml"), config) {
+        Ok(ws) => {
+            // Because we only inherit the package field of workspace for valid members of workspace,
+            // we did not inherit it directly above.
+            let root_manifest = paths::read(ws.root_manifest())?;
+            let workspace_document =
+                parse_document(root_manifest.as_str(), ws.root_manifest(), config)?;
+            if let Some(workspace_package_keys) = workspace_document
+                .get("workspace")
+                .and_then(|workspace| workspace.get("package"))
+                .and_then(|package| package.as_table())
+            {
+                // Because we need to keep the format of the `Cargo.toml` file,
+                // we can not use `toml_edit`/`toml` to modify the `Cargo.toml` file directly.
```

Comment:
Why can't `toml_edit`/`toml` be used to modify the `Cargo.toml`

Reply:
There shouldn't be; we should be modifying the file with `toml_edit` rather than using string interpolation

Also, editing after-the-fact being more difficult was one of the potential problems I raised with doing the two-write approach

Reply:
> Why can't `toml_edit`/`toml` be used to modify the `Cargo.toml`



> There shouldn't be; we should be modifying the file with `toml_edit` rather than using string interpolation

But if we use `toml_edit`/`toml` it will other the keys alphabetically. But do we want to keep the name first?

Reply:
`toml` is dependent on the data structure used.

`toml_edit` is format preserving and is what `cargo add` and `cargo rm` use.

Reply:
> `toml` is dependent on the data structure used.
> 
> `toml_edit` is format preserving and is what `cargo add` and `cargo rm` use.

Got it. Thanks for your explanation.

But I think the current approach is more reusable and easier. In the future, if we want to add the package to the members atomically, we only need to call `create_manifest_with_inherited_workspace_package_fields`. If we use `toml_edit` to edit the existing file, it will make the second write couple with the first write. But maybe in the future, we will remove the first write and only keep the second one.

Reply:
Updated the comment.

Reply:
I consider using `toml_edit` a blocker for merging.

Reply:
I will give it a try:(

Reply:
Updated!
@epage Do you think it would be beneficial to use 'toml_edit' when creating a new 'Cargo.toml'?